src/software/thunderscope/robot_diagnostics/chicker.py

```diff
@@ -73,6 +89,71 @@ def __init__(self, proto_unix_io):
         self.geneva_value = 3
         self.power_value = 1
 
+    def kick_clicked(self):
+        # if button is enabled
+        if self.kick_button_enable:
+            # send kick primitive
+            self.send_kick_or_chip(True)
+            self.toggle_kick_button()
+
+            # set and start timer to re-enable kick button after 3 seconds
+            t = Timer(3, lambda s: s.toggle_kick_button(), [self])
+            t.start()
+
+    def chip_clicked(self):
+        # if button is enabled
+        if self.chip_button_enable:
+            # send chip primitive
+            self.send_kick_or_chip(False)
+            self.toggle_chip_button()
+
+            # set and start timer to re-enable chip button after 3 seconds
+            t = Timer(3, lambda s: s.toggle_chip_button(), [self])
+            t.start()
+
+    def toggle_kick_button(self):
+        self.kick_button_enable = not self.kick_button_enable
+
+    def toggle_chip_button(self):
+        self.chip_button_enable = not self.chip_button_enable
+
+    def buttons_disable(self):
+        self.kick_button_enable = False
+        self.chip_button_enable = False
+
+    def buttons_enable(self):
+        self.kick_button_enable = True
+        self.chip_button_enable = True
+
+    def send_kick_or_chip(self, kick):
+        """Sends a kick or chip primitive dependent on boolean parameter
+
+        :param kick: boolean to indicate whether to send Kick (True) or Chip (False) primitive
+        :returns: None
+
+        """
+
+        # gets slider values and sets label to that value
+        geneva_value = self.geneva_slider.value()
+        self.geneva_label.setText(Slot.Name(geneva_value))
+
+        power_value = self.power_slider.value()
+        self.power_label.setText(str(power_value))
```

Comment:
I believe this should still live in the refresh function otherwise the UI for the slider value wont update until a button is clicked/ enabled